docs/source/tutorial/tutorial-query-ui.md

```diff
@@ -162,126 +142,93 @@ override func viewDidLoad() {
+> **Note**: You're also unwrapping the `missionNameLabel` because even though it's an Implicitly Unwrapped Optional, it won't be present if `configureView` is called before `viewDidLoad`.
 
-Next, back in `MasterViewController.swift`, update the `prepareForSegue` method to obtain the most recently selected row and pass its corresponding launch details to the detail view controller: 
+Next, back in `LaunchesViewController.swift`, update the `prepareForSegue` method to obtain the most recently selected row and pass its corresponding launch details to the detail view controller. Replace the `TODO` and below with the following: 
 
-```swift:title=MasterViewController.swift
-override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-  guard let selectedIndexPath = self.tableView.indexPathForSelectedRow else {
-    // Nothing is selected, nothing to do
-    return
-  }
-    
-  guard let listSection = ListSection(rawValue: selectedIndexPath.section) else {
-    assertionFailure("Invalid section")
-    return
-  }
+```swift:title=LaunchesViewController.swift
+guard let selectedIndexPath = self.tableView.indexPathForSelectedRow else {
+  // Nothing is selected, nothing to do
+  return
+}
     
-  switch listSection {
-  case .launches:
-    guard
-      let destination = segue.destination as? UINavigationController,
-      let detail = destination.topViewController as? DetailViewController else {
-        assertionFailure("Wrong kind of destination")
-        return
-    }
+guard let listSection = ListSection(rawValue: selectedIndexPath.section) else {
+  assertionFailure("Invalid section")
+  return
+}
     
-    let launch = self.launches[selectedIndexPath.row]
-    detail.launchID = launch.id
-    self.detailViewController = detail
+switch listSection {
+case .launches:
+  guard
+    let destination = segue.destination as? UINavigationController,
+    let detail = destination.topViewController as? DetailViewController else {
+      assertionFailure("Wrong kind of destination")
+      return
   }
-}
-```
-
-One slightly unrelated detail: In `SceneDelegate.swift`, one line relies on the `detailItem` property you deleted earlier, which now produces an error. Replace the erroring line with:
 
-```swift:title=SceneDelegate.swift
-if topAsDetailController.launchID == nil {
+  let launch = self.launches[selectedIndexPath.row]
+  detail.launchID = launch.id
+  self.detailViewController = detail
+}
 ```
 
 Build and run, and tap on any of the launches. You'll now see the launch ID for the selected launch when you land on the page:
 
-
-<img src="images/detail_view_launch_id.png" alt="Empty screen with launch ID" class="screenshot" width="300">
-</img>
+<img src="images/detail_view_launch_id.png" alt="Empty screen with launch ID" class="screenshot" width="300"/>
```

Comment:
Following the tutorial I'm getting stuck here and I suspect there is a missing step re. the `shouldPerformSegue(withIdentifier:sender:)` method; there is still a TODO in that method with the following `deselectRow` causing the problem.

Reply:
Fixed by [removing row deselection from the starter](https://github.com/apollographql/iOSTutorial/commit/5a412900d0c0ce4f5d224bf081d5834cf56ab79e) - thanks for this, I'd meant to change that and thought I already had!